### PR TITLE
[WIP] Posting Schedule state fixes

### PR DESCRIPTION
**Pull request by MayaUribe**
### Purpose
Doing some changes and a bit of a refactor to fix state saving for Posting Schedule.

#### Staging Deployment
To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/posting-schedule/index.test.jsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import {
   days,
   timezones,
 } from './components/PostingSchedule/settingsData';
+import GridList from '@bufferapp/publish-shared-components/GridList/story';
```
Comment (karelm) [inline packages/posting-schedule/index.test.jsx]: Hey @MayaUribe this looks like an unused line of code here, would you mind taking a look?

File: packages/posting-schedule/reducer.js [hunk 2]
```diff
@@ -171,20 +193,67 @@ export default (state = initialState, action) => {
...
+    case actionTypes.REMOVE_SCHEDULE_TIME:
+    case actionTypes.ADD_SCHEDULE_TIME:
+    case actionTypes.UPDATE_SCHEDULE_TIME:
+      profileId = getProfileId(action);
+      if (profileId) {
+        return {
+          profileId,
+          byProfileId: {
+            ...state.byProfileId,
+            [profileId]: profileReducer(state.byProfileId[profileId], action),
```
Comment (karelm) [inline packages/posting-schedule/reducer.js]: 😍

Comment (karelm) [reply packages/posting-schedule/reducer.js]: do you not need to spread in the rest of state here too? 
...state ?

File: packages/posting-schedule/index.test.jsx (hunk 1 above)
Comment (MayaUribe) [reply packages/posting-schedule/index.test.jsx]: Deleting :)

File: packages/posting-schedule/reducer.js (hunk 2 above)
Comment (MayaUribe) [reply packages/posting-schedule/reducer.js]: Oh yep I forgot about that

**Pull request closed without merging**